chemclipse/plugins/org.eclipse.chemclipse.chromatogram.xxd.peak.detector.supplier.firstderivative/src/org/eclipse/chemclipse/chromatogram/xxd/peak/detector/supplier/firstderivative/core/PeakDetectorMSD.java

```diff
@@ -159,12 +165,12 @@ private void buildAndStorePeaks(List<IRawPeak> rawPeaks, IChromatogramMSD chroma
 	 * @param windowSize
 	 * @return {@link IFirstDerivativeDetectorSlopes}
 	 */
-	public static IFirstDerivativeDetectorSlopes getFirstDerivativeSlopes(IChromatogramSelectionMSD chromatogramSelection, WindowSize windowSize) {
+	public static IFirstDerivativeDetectorSlopes getFirstDerivativeSlopes(IChromatogramSelectionMSD chromatogramSelection, WindowSize movingAverageWindowSize, Collection<Number> filterIons, FilterMode mode) {
```

Comment:
Why not pass in the result of getIonFilter(filterIons, mode) as a parameter? Would be more flexible IMO

Reply:
resolved with c8e91a67951b0ae0f5a484f48a1907c6b6fb45e3.